src/api/courses.js

```diff
@@ -69,6 +143,74 @@ router.get(
   })
 )
 
+// Get course queue data
+router.get(
+  '/:courseId/data/questions',
+  [requireCourseStaff, requireCourse, failIfErrors],
+  safeAsync(async (req, res, _next) => {
+    const { id: courseId } = res.locals.course
+    const questions = await Question.findAll({
+      include: [
+        {
+          model: Queue,
+          include: [
+            {
+              model: Course,
+              attributes: [['name', 'CourseName']],
+              required: true,
+              where: { id: Sequelize.col('queue.courseId') },
+            },
+          ],
+          attributes: [
+            ['id', 'queueId'],
+            'courseId',
+            ['name', 'QueueName'],
+            ['location', 'QueueLocation'],
+            ['createdAt', 'Queue_CreatedAt'],
+          ],
+          required: true,
+          where: { courseId, id: Sequelize.col('question.queueId') },
+        },
+        {
+          model: User,
+          as: 'askedBy',
+          attributes: [
+            ['netid', 'AskedBy_netid'],
+            ['universityName', 'AskedBy_UniversityName'],
+          ],
+          required: true,
+          where: { id: Sequelize.col('question.askedById') },
+        },
+        {
+          model: User,
+          as: 'answeredBy',
+          attributes: [
+            ['netid', 'AnsweredBy_netid'],
+            ['universityName', 'AnsweredBy_UniversityName'],
+          ],
+          required: false,
+          where: { id: Sequelize.col('question.answeredById') },
+        },
+      ],
+      attributes: [
+        'id',
+        'topic',
+        'enqueueTime',
+        'dequeueTime',
+        'answerStartTime',
+        'answerFinishTime',
+        'comments',
+        'preparedness',
+        ['location', 'UserLocation'],
+      ],
+      order: [['enqueueTime', 'DESC']],
+      raw: true,
+    })
+
+    res.send(getCsv(questions))
```

Comment:
The response should have a `Content-Type` of `text/csv`. It could also include a `Content-Disposition` specifying that the response is an attachment and should be downloaded instead of being displayed in the browser. More details: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition